plugin/evm/vm_test.go

```diff
@@ -153,10 +153,6 @@ func setupGenesis(
 	atomicMemory := atomic.NewMemory(prefixdb.New([]byte{0}, baseDB))
 	ctx.SharedMemory = atomicMemory.NewSharedMemory(ctx.ChainID)
 
-	// NB: this lock is intentionally left locked when this function returns.
-	// The caller of this function is responsible for unlocking.
-	ctx.Lock.Lock()
```

Comment:
uh this was a weird assumption. This might work for coreth (as it's expected a lock to be held in engine before calling VM functions) but this does not work here.